src/Application/GameConfig.h

```diff
@@ -4,13 +4,1474 @@
+            }
+
+            inline bool DefaultNoMargareth() {
+                return false;
+            }
+            inline void SetNoMargareth(bool is_enabled) {
+                this->no_margareth = is_enabled;
+            }
+            inline bool GetNoMargareth() {
+                return this->no_margareth;
+            }
+            inline void ToggleNoMargareth() {
+                this->no_margareth = !this->no_margareth;
+            }
+
+            inline bool DefaultVerboseLogging() {
+                return false;
+            }
+            inline void SetVerboseLogging(bool is_enabled) {
+                this->verbose_logging = is_enabled;
+            }
+            inline bool GetVerboseLogging() {
+                return this->verbose_logging;
+            }
+            inline void ToggleVerboseLogging() {
+                this->verbose_logging = !this->verbose_logging;
+            }
+
+         private:
+            /** Enable all available spells for each character in spellbook bypassing all class restrictions. Currently also all skills will behave like they are on GM level. */
+            bool all_magic;
+
+            bool infinite_food;
+            bool infinite_gold;
+
+            /** Debug lightmap and decals outlines. */
+            bool lightmaps_decals;
+
+            /** Draw BLV portal frames. */
+            bool portal_outlines;
+
+            bool terrain;
+            bool town_portal;
+
+            /** Increase party movement speed by 12x. Most likely you want to use that option with no_damage option enabled as collision physics often will shoot you in the air. */
+            bool turbo_speed;
+
+            /** Game will behave like spell wizard eye is casted and it will never expire. */
+            bool wizard_eye;
+
+            bool show_fps;
+
+            /** Face pointed by mouse will flash with red for buildings or green for indoor. */
+            bool show_picked_face;
+
+            /** Disable all actors */
+            bool no_actors;
+
+            /** Disable all incoming damage to party. */
+            bool no_damage;
+
+            /** Disable all decorations */
+            bool no_decorations;
+
+            /** Skip intro movie on startup */
+            bool no_intro;
+
+            /** Skip 3do logo on startup */
+            bool no_logo;
+
+            /** Disable Margareth's tour messages on Emerald Island. */
+            bool no_margareth;
+
+            /** Don't play any sounds. Currently it doesn't affect in-house movies. */
+            bool no_sound;
+
+            /** Don't play any movies. */
+            bool no_video;
+
+            bool verbose_logging;
+        };
+
+        Debug debug;
+
+        class Gameplay {
+         public:
+            const std::string section_name = "gameplay";
+
+            void Default();
+            void Load();
+            void Save();
+
+            inline int DefaultMaxFlightHeight() {
```

Comment:
I kinda have the same feeling here.

static constexpr int DefaultMaxFlightHeight = 4000;
validated_value<int, &ValidateMaxFlightHeight> max_flight_height = DefaultMaxFlightHeight;